src/toolkits/drawing_classifier/drawing_classifier.cpp

```diff
@@ -506,7 +506,10 @@ void drawing_classifier::iterate_training() {
   if (validation_data_iterator_) {
     add_or_update_state({
         {"validation_accuracy", average_val_accuracy},
-        {"validation_log_loss", average_val_loss},
+    });
+  } else {
+    add_or_update_state({
```

Comment:
probably we don't want the user to be aware of the validation metrics if we don't have validation data set.

Reply:
But it adds it to the API exposed via Python so it's better for the API to be consistent irrespective of the options that are used during the training process.